### ISSUE TITLE
Classes naming convention

### ISSUE THREAD

**Issue by neonailol** [yegor256/cactoos#913]
Why classes from different packages have same names?
e.g. there are `Filtered` class in `collection`, `list`, `iterator` and `iterable`
and when you try to use different `Filtered` classes in one file you end up with this clutter:

```kotlin
val collection = CollectionOf(1,2,3)
Filtered(FuncOf(true), collection)
org.cactoos.iterable.Filtered(FuncOf(false), collection)
org.cactoos.iterator.Filtered(FuncOf(false), collection)
```

Maybe these names should be questioned? 
For example name them based on the interface they implement `FilteredList`, `FilteredCollection`, `FilteredIterable`, `FilteredIterator`? and maybe hide iterable and iterator classes from public API?

**Comment by 0crat** [comment]
@llorllale[/z](https://www.0crat.com/u/llorllale) please, pay attention to this issue

**Comment by 0crat** [comment]
@neonailol[/z](https://www.0crat.com/u/neonailol) this project will fix the problem faster if you donate a few dollars to it; just [click here](https://www.0crat.com/contrib/C63314D6Z) and pay via Stripe, it's very fast, convenient and appreciated; thanks a lot!

**Comment by llorllale** [comment]
@neonailol I think avoiding suffixes is ideal. The class' package provides sufficient context. (I have been pondering removing the `Text` suffix from `org.cactoos.text` for some time now)

Can you show an example of the "clutter" this convention is causing you? (let's keep in mind that [we should ideally have just one statement - a 'return' - per method](https://www.yegor256.com/2015/08/18/multiple-return-statements-in-oop.html))

**Comment by neonailol** [comment]
@llorllale Then you have inconsistency in naming: there are `SolidCollection`, `SolidList`, `SolidMap` and few others.
And we do not live in an ideal world, and as the old saying goes: 'In theory, there is no difference between theory and practice. But, in practice, there is.'

**Comment by llorllale** [comment]
@neonailol yes, we do have naming inconsistencies.

This ideal is worth pushing for, as it will hopefully drive the design of user classes to a more [elegant] object-oriented style.

**Comment by neonailol** [comment]
@llorllale so if we go to a short convention and have `Solid` class in `map` package, and `Solid` class in `list` package, when I try to have to write code that uses both of them, I have to use the long name with a full package in it for one of these, this seems not very practical to me. And I can't think that can be solved in an elegant way.

**Comment by 0crat** [comment]
@llorllale[/z](https://www.0crat.com/u/llorllale) not enough funds available in the project, can't set budget of job [#913](https://github.com/yegor256/cactoos/issues/913), see [§21](http://www.zerocracy.com/policy.html#21); @llorllale[/z](https://www.0crat.com/u/llorllale) will get no money on completion; in order to fix that, add funds to the project and assign the job again

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#913` already assigned to @llorllale, can't assign to @llorllale

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 point(s) just awarded to @neonailol[/z](https://www.0crat.com/u/neonailol)

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by llorllale** [comment]
@neonailol yes, a choice has to be made: either we use suffixes everywhere or we remove them all where possible[1]

a) Use suffixes everywhere
**Pros**
1. Flatter learning curve for new users of `cactoos` and the general philosophy behind our designs
2. Avoids name clashes in [what I believe are] less common use cases such as yours

**Cons**
1. Have to type long suffixes, probably multiple times if we consider how focused our classes are. Example: `new FilteredCollection<>(e -> ..., new JoinedCollection<>(new HeadOfCollection<>(10, col1), new TailOfCollection<>(10, col2)));`. Additionally, you may have typed "Collection" at least once each for `col1` and `col2`
2. Excessive redundancy (see previous example), affecting readability when you compose objects of the same type
3. Excessive redundancy for most scenarios where you're composing objects of different types, such as your case. Eg.: you generally don't need to explicitly filter both an `Iterable` *and* its `Iterator`.

b) Removing all suffixes
This would basically flip all cons of `a)` into pros and vice-versa. I will say though that as far as the learning curve for newcomers goes: i) it's not a big deal, and ii) we are already a magnet for programmers that are willing to try new paradigms.

--------------------
[1] Keep them to avoid name clashes in the same package and where there isn't a better name for either class

**Comment by SilasReinagel** [comment]
From my experience with name collisions, I vote on adding Suffixes. This also has the added benefit of making all objects nouns again. `Filtered` is an adjective. `FilteredText` is a noun.

However, using generics instead of having special filters for each might eliminate the need for various `Filtered` implementations (and other functional adjective operations) completely.

**Comment by Vatavuk** [comment]
@llorllale I'm highly against suffixes. I'm heavily user of cactoos (In all my projects) using
`filtered`, `mapped` etc. all the time.  I've never run in a situation that I need, for instance,  filtered collection and filtered iterable in the same class. But this again depends on the situation, of course. Using such primitive objects is already verbose and adding suffixes would make composition of these objects worse, messy and just ugly. 
I think the current naming situation is fine except for the few `*Of` classes that could be merged into one.
If I need to choose between those two I would choose a) but think about changing these
sufix free object names to something more meaningful than just for example `Solid`

**Comment by Vatavuk** [comment]
@llorllale Just look at this:
```java
new FilteredCollection<>(
    e -> ...,
    new JoinedCollection<>(
        new HeadOfCollection<>(10, col1),
        new TailOfCollection<>(10, col2))
    )
)
```
This Collection suffix just adds noise. You have to focus more to remove
the noise in your head and to see what the code does.

And now look at this:
```java
new Filtered<>(
    e -> ...,
    new Joined<>(
        new HeadOf<>(10, col1),
        new TailOf<>(10, col2))
    )
)
```
Pure and simple, and way faster to digest, in my opinion.

**Comment by neonailol** [comment]
Also, another source of confusion is what is mutable and what is immutable, should it be seen when you look at the code or not?

**Comment by Vatavuk** [comment]
@neonailol do we have mutable things in cactoos?

**Comment by neonailol** [comment]
@Vatavuk `CollectionNoNulls` and some other `*NoNulls`  does not explicitly immutable, it delegates everything to an encapsulated collection, which can be of any type, and I may be just nitpicking here.

**Comment by neonailol** [comment]
@llorllale I think this example is extrapolated too much

```
new FilteredCollection<>(
    e -> ...,
    new JoinedCollection<>(
        new HeadOfCollection<>(10, col1),
        new TailOfCollection<>(10, col2))
    )
)
```

Because looking through the code, there is `FuncOf` instead of `FunctionOf` for example and many other different shorthands. So `FilteredCollection` can be shrunk to `FilteredColl` or smth better. When this pattern is used with lists, it does not look so bad

```
new FilteredList<>(
    e -> ...,
    new JoinedList<>(
        new HeadOfList<>(10, col1),
        new TailOfList<>(10, col2))
    )
)
```

**Comment by llorllale** [comment]
@SilasReinagel 

> I vote on adding Suffixes. This also has the added benefit of making all objects nouns again. Filtered is an adjective. FilteredText is a noun.

Using [nominal adjectives](https://en.wikipedia.org/wiki/Nominalized_adjective) brings us almost back to nouns again. The wikipedia article has a nice example based on which I ask you: do you commonly speak the phrase as "the rich people and the poor people" or as "the rich and the poor"? I bet you use the latter. A very similar thing is going on here.

Consider that objects don't exist in isolation and instead they appear in some *context* (hopefully just the previous few lines of code). In that case, why repeat the obvious? If I already know that `names` is an `Iterable`, then isn't it obvious what's going on in the following?

```java
new Filtered<>(n -> n.startsWith("Silas"), col1).
```

**Comment by llorllale** [comment]
@neonailol 

> there is FuncOf instead of FunctionOf

It's because the interface is named [`Func`](https://github.com/yegor256/cactoos/blob/92b77ed9e2d34ddec1722393c36d3659c7bd1780/src/main/java/org/cactoos/Func.java#L49)

**Comment by neonailol** [comment]
@llorllale And an interface named `Func` instead of `Function`

**Comment by SilasReinagel** [comment]
@llorllale In my personal experience, brevity is good, but namespace collisions are more harmful than brevity is helpful. Codebases with namespace collisions are very hard to work in. IDEs work against you when there are enough collisions.

Nominalized adjectives are fine, although they are harder to understand in a codebase when viewed separate from their usage sites.

**Comment by Vatavuk** [comment]
@SilasReinagel @neonailol @llorllale I dont think that naming collision is a real world scenario.
Im talking only about iterable, collection and list family (filtered, mapped and so on).

Is there a point of using filtered (collection) and filtered (iterable) in the same class?

We can import `Filtered` (collection) and use it like this:
```java
Collection<String> strings = new Filtered(e -> ..., stringCollection);
Iterable<String> iterable = new Filtered(e -> ..., stringCollection);
```
If I am using` Filtered`(Iterable) somewhere in my code and I need collection I just import `Filtered`(Collection) instead of `Filtered`(Iterable). And if I have `Filtered`(collection) and need `Iterable` I can use already imported `Filtered`. Same for the list.
We are not solving the problem we are creating it:).

**Comment by Vatavuk** [comment]
@neonailol @llorllale @SilasReinagel 
One more example, using suffixes:
```java
new FilteredColl<>(
    e -> ...,
    new JoinedIterable<>(
        new MappedList<>(e-> ..., list1),
        new MappedList <>(e->..., list2))
    )
) 
```
Do I really need to care which underlying interfaces are used
to construct the result? I only care about the result, please do not show me this noise, because than I need to process why are we using list or collection or whatever. We should 
protect ourselves from this additional complexity

**Comment by vizotov** [comment]
@neonailol @llorllale @SilasReinagel @Vatavuk 

Why do you need classes with the same name in different packages? To get the right interface on the 
output?
Maybe it would be better to use a decorator instead?

Then this code:

```java
new FilteredCollection<>(
    e -> ...,
    new JoinedCollection<>(
        new HeadOfCollection<>(10, col1),
        new TailOfCollection<>(10, col2))
    )
)
```

will look like this:
```java
new CollectionOf<>(
      new Filtered<>(
          e -> ...,
      new Joined<>(
          new HeadOf<>(10, col1),
          new TailOf<>(10, col2))
      )
    )
```

or if we need Iterable:

```java
new IterableOf<>(
      new Filtered<>(
          e -> ...,
      new Joined<>(
          new HeadOf<>(10, col1),
          new TailOf<>(10, col2))
      )
    )
```

IMO:
1. There should be no classes with the same names in different packages.
2. Instead of suffixes, decorators should be used.

**Comment by Vatavuk** [comment]
@vizotov What interface does `Filtered` and `Joined` implements in your example? I don't understand how you solved the issue? We are using multiple `Filtered` objects just to give users more flexibility. But in the end, only one of those objects will be used per a class. I don't see a scenario when we need them both.

**Comment by vizotov** [comment]
@Vatavuk 
These classes can to implement `Iterable`, as is more generic in comparison with `Collection`. Or we will have to create an even more general interface. All relationships between these objects must be implemented using constructors. The use of only one of those objects per a class is an artificial restriction. If we can do without this restriction, why not?

When using decorators flexibility will not be worse. Instead of importing `Filtered` classes from different packages, we will decorate the output.

**Comment by Vatavuk** [comment]
@vizotov 
Ok I think I understand now. This means that we will expand current `Filtered` (that implements `Iterable`)
with additional ctors that takes `Collection` and `List` and remove other `Filtered` objects. than we use decorators for different outputs.
I totaly agree. @llorllale This seems to me as the best approach.

**Comment by vizotov** [comment]
@Vatavuk 
Yeah, that's what I meant.

**Comment by llorllale** [comment]
@Vatavuk @vizotov of all the classes with the same names, the ones from which package do we keep?

**Comment by Vatavuk** [comment]
@llorllale I would leave those from `iterable` and `iterator` packages and would add suffixes `*Iterator` for `iterator` family.

**Comment by Vatavuk** [comment]
@llorllale The option of leaving only `iterator` package is also fine with me.

**Comment by llorllale** [comment]
@Vatavuk @vizotov allow me a little time to think it over

**Comment by llorllale** [comment]
@vizotov @Vatavuk so we are now discussing two issues here:

1. Naming convention
2. Remove "unnecessary" classes

@vizotov can you please open an issue for the second point?

With that said, I think you both agree on removing unnecessary suffixes like "Collection", "List", "Text", etc?

**Comment by Vatavuk** [comment]
@llorllale Yes I vote for removing suffixes

**Comment by 0crat** [comment]
@llorllale[/z](https://www.0crat.com/u/llorllale) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by vizotov** [comment]
@llorllale 
Issue #915 is created.

Create an issue for removing suffixes?

**Comment by llorllale** [comment]
@vizotov we'll handle suffixes in this issue ("class naming convention")

**Comment by Zefick** [comment]
FYI one more possible solution of the problem is factory classes:

```
import org.cactoos.factory.Filtered;
import org.cactoos.factory.Joined;
import org.cactoos.factory.Head;
import org.cactoos.factory.Tail;

Filtered.from(
    e -> ...,
    Joined.from(
        Head.of(10, col1),
        Tail.of(10, col2))
    )
)
```
I understand that the chances of accepting this approach are too low because of library's principle but it's the best you could do.

**Comment by llorllale** [comment]
@yegor256 @Zefick @Vatavuk @SilasReinagel @vizotov @neonailol 

Let's codify this part of our naming convention. Here is what I've got:

> Decorators use a [nominalized adjective](https://en.wikipedia.org/wiki/Nominalized_adjective) as their name where technically possible.
> 
> **Rationale**
> [Elegant](www.elegantobjects.org) code is declarative and tends to be written as succinctly as possible (1 "return" statement per method is ideal). Therefore, its context is very compact, making suffixes (or prefixes) generally not required.
>
> Example:
> You're implementing this interface:
> ```java
> interface LiveProducts extends Iterable<Product> {
>    ...
> }
> ```
> Assume your implementation is given an `Iterable<Product>` as a parameter (referenced as `products`).
> Then it is assumed that the following implementation of `LiveProducts` is perfectly readable:
> ```java
> public Iterator<Product> iterator() {
>     return new IteratorOf<>(
>         new Filtered<>(
>             p -> Product.ACTIVE.equals(p.status()),
>             this.products
>         )
>     );
> }
> ```
 

These are the changes I think should happen:

Before|After
----|----
org.cactoos.bytes.HexBytes|org.cactoos.bytes.HexOf
org.cactoos.collection.SolidCollection|org.cactoos.collection.Solid
org.cactoos.collection.StickyCollection|org.cactoos.collection.Sticky
org.cactoos.collection.SyncCollection|org.cactoos.collection.Synced
org.cactoos.func.AsyncFunc|org.cactoos.func.Async
org.cactoos.func.ChainedFunc|org.cactoos.func.Chained
org.cactoos.func.RepeatedFunc|org.cactoos.func.Repeated
org.cactoos.func.RetryFunc|org.cactoos.func.Retry
org.cactoos.func.TimedFunc|org.cactoos.func.Timed
org.cactoos.io.StickyInput|org.cactoos.io.Sticky
org.cactoos.io.TailInput|org.cactoos.io.TailOf
org.cactoos.iterable.SolidIterable|org.cactoos.iterable.Solid
org.cactoos.iterable.StickyIterable|org.cactoos.iterable.Sticky
org.cactoos.iterable.SyncIterable|org.cactoos.iterable.Synced
org.cactoos.iterator.SyncIterator|org.cactoos.iterator.Synced
org.cactoos.list.StickyList|org.cactoos.list.Sticky
org.cactoos.list.SyncList|org.cactoos.list.Synced
org.cactoos.map.SolidMap|org.cactoos.map.Solid
org.cactoos.map.StickyMap|org.cactoos.map.Sticky
org.cactoos.map.SyncMap|org.cactoos.map.Synced
org.cactoos.scalar.CheckedScalar|org.cactoos.scalar.Checked
org.cactoos.scalar.IoCheckedScalar|org.cactoos.scalar.IoChecked
org.cactoos.scalar.RetryScalar|org.cactoos.scalar.Retry
org.cactoos.scalar.SolidScalar|org.cactoos.scalar.Solid
org.cactoos.scalar.StickyScalar|org.cactoos.scalar.Sticky
org.cactoos.scalar.SumOfDoubleScalar|org.cactoos.scalar.SumOfDouble
org.cactoos.scalar.SumOfFloatScalar|org.cactoos.scalar.SumOfFloat
org.cactoos.scalar.SumOfIntScalar|org.cactoos.scalar.SumOfInt
org.cactoos.scalar.SumOfLongScalar|org.cactoos.scalar.SumOfLong
org.cactoos.scalar.SyncScalar|org.cactoos.scalar.Synced
org.cactoos.scalar.UncheckedScalar|org.cactoos.scalar.Unchecked
all `*Text` classes|all of them

**Comment by llorllale** [comment]
@0crat waiting

**Comment by 0crat** [comment]
> @0crat waiting ([here](https://github.com/yegor256/cactoos/issues/913#issuecomment-402332433))

@llorllale The impediment for [#913](https://github.com/yegor256/cactoos/issues/913) was registered successfully by @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by 0crat** [comment]
> @0crat waiting ([here](https://github.com/yegor256/cactoos/issues/913#issuecomment-402332433))

@llorllale Job [#913](https://github.com/yegor256/cactoos/issues/913) is not in the agenda of @llorllale[/z](https://www.0crat.com/u/llorllale), can't set impediment

**Comment by neonailol** [comment]
> Elegant code is declarative and tends to be written as succinctly as possible (1 "return" statement per method is ideal). Therefore, its context is very compact, making suffixes (or prefixes) generally not required.

This comes from the idea that we must have one class per file, but i don't see why it should be that way, i tend to put close related classes in one file, and this is where name clashes can happen, in java we have restriction, that file can have only one public class, i.e. this is legal code:
```java
package sample;
public class Foo {}
class Bar {}
```
But I mainly write in Koltin where are no such restrictions, and one can put any number of classes in one file.

**Comment by llorllale** [comment]
@neonailol it comes from the idea that cohesive methods tend to do just one task, and thus tend to have the least amount of statements.

**Comment by llorllale** [comment]
@neonailol @Vatavuk @vizotov @Zefick are there any objections to [this](https://github.com/yegor256/cactoos/issues/913#issuecomment-402332247) proposal?

I want this issue closed by friday. Thanks in advance!

**Comment by Vatavuk** [comment]
@llorllale this is fine with me

**Comment by vizotov** [comment]
@llorllale no objections.

**Comment by pinaf** [comment]
@llorllale Hi. Is this issue on hold?

**Comment by 0pdd** [comment]
@neonailol the puzzle [#958](https://github.com/yegor256/cactoos/issues/958) is still not solved.

**Comment by llorllale** [comment]
@pinaf let's make it happen. I've already started in 1dfd0766c34fe3c3255b1a8cbba318cf383d08c8; there's open puzzle #958 to continue

**Comment by 0crat** [comment]
Job [#913](https://github.com/yegor256/cactoos/issues/913) is not in the agenda of @llorllale[/z](https://www.0crat.com/u/llorllale), can't retrieve data and time of add

**Comment by 0crat** [comment]
The job [#913](https://github.com/yegor256/cactoos/issues/913) is now out of scope

**Comment by 0crat** [comment]
Order was finished: +30 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by 0pdd** [comment]
@neonailol the puzzle [#964](https://github.com/yegor256/cactoos/issues/964) is still not solved; solved: [#958](https://github.com/yegor256/cactoos/issues/958).

**Comment by 0pdd** [comment]
@neonailol the puzzle [#980](https://github.com/yegor256/cactoos/issues/980) is still not solved; solved: [#958](https://github.com/yegor256/cactoos/issues/958), [#964](https://github.com/yegor256/cactoos/issues/964).

**Comment by 0pdd** [comment]
@neonailol 5 puzzles [#1003](https://github.com/yegor256/cactoos/issues/1003), [#1004](https://github.com/yegor256/cactoos/issues/1004), [#1005](https://github.com/yegor256/cactoos/issues/1005), [#1006](https://github.com/yegor256/cactoos/issues/1006), [#1007](https://github.com/yegor256/cactoos/issues/1007) are still not solved; solved: [#958](https://github.com/yegor256/cactoos/issues/958), [#964](https://github.com/yegor256/cactoos/issues/964), [#980](https://github.com/yegor256/cactoos/issues/980).

**Comment by 0pdd** [comment]
@neonailol 4 puzzles [#1003](https://github.com/yegor256/cactoos/issues/1003), [#1005](https://github.com/yegor256/cactoos/issues/1005), [#1006](https://github.com/yegor256/cactoos/issues/1006), [#1025](https://github.com/yegor256/cactoos/issues/1025) are still not solved; solved: [#1004](https://github.com/yegor256/cactoos/issues/1004), [#1007](https://github.com/yegor256/cactoos/issues/1007), [#958](https://github.com/yegor256/cactoos/issues/958), [#964](https://github.com/yegor256/cactoos/issues/964), [#980](https://github.com/yegor256/cactoos/issues/980).

**Comment by 0pdd** [comment]
@neonailol 4 puzzles [#1003](https://github.com/yegor256/cactoos/issues/1003), [#1005](https://github.com/yegor256/cactoos/issues/1005), [#1025](https://github.com/yegor256/cactoos/issues/1025), [#1046](https://github.com/yegor256/cactoos/issues/1046) are still not solved; solved: [#1004](https://github.com/yegor256/cactoos/issues/1004), [#1006](https://github.com/yegor256/cactoos/issues/1006), [#1007](https://github.com/yegor256/cactoos/issues/1007), [#958](https://github.com/yegor256/cactoos/issues/958), [#964](https://github.com/yegor256/cactoos/issues/964), [#980](https://github.com/yegor256/cactoos/issues/980).

**Comment by 0pdd** [comment]
@neonailol 5 puzzles [#1003](https://github.com/yegor256/cactoos/issues/1003), [#1025](https://github.com/yegor256/cactoos/issues/1025), [#1046](https://github.com/yegor256/cactoos/issues/1046), [#1064](https://github.com/yegor256/cactoos/issues/1064), [#1065](https://github.com/yegor256/cactoos/issues/1065) are still not solved; solved: [#1004](https://github.com/yegor256/cactoos/issues/1004), [#1005](https://github.com/yegor256/cactoos/issues/1005), [#1006](https://github.com/yegor256/cactoos/issues/1006), [#1007](https://github.com/yegor256/cactoos/issues/1007), [#958](https://github.com/yegor256/cactoos/issues/958), [#964](https://github.com/yegor256/cactoos/issues/964), [#980](https://github.com/yegor256/cactoos/issues/980).

**Comment by 0pdd** [comment]
@neonailol 5 puzzles [#1003](https://github.com/yegor256/cactoos/issues/1003), [#1025](https://github.com/yegor256/cactoos/issues/1025), [#1046](https://github.com/yegor256/cactoos/issues/1046), [#1064](https://github.com/yegor256/cactoos/issues/1064), [#1074](https://github.com/yegor256/cactoos/issues/1074) are still not solved; solved: [#1004](https://github.com/yegor256/cactoos/issues/1004), [#1005](https://github.com/yegor256/cactoos/issues/1005), [#1006](https://github.com/yegor256/cactoos/issues/1006), [#1007](https://github.com/yegor256/cactoos/issues/1007), [#1065](https://github.com/yegor256/cactoos/issues/1065), [#958](https://github.com/yegor256/cactoos/issues/958), [#964](https://github.com/yegor256/cactoos/issues/964), [#980](https://github.com/yegor256/cactoos/issues/980).

**Comment by 0pdd** [comment]
@neonailol 4 puzzles [#1003](https://github.com/yegor256/cactoos/issues/1003), [#1025](https://github.com/yegor256/cactoos/issues/1025), [#1046](https://github.com/yegor256/cactoos/issues/1046), [#1074](https://github.com/yegor256/cactoos/issues/1074) are still not solved; solved: [#1004](https://github.com/yegor256/cactoos/issues/1004), [#1005](https://github.com/yegor256/cactoos/issues/1005), [#1006](https://github.com/yegor256/cactoos/issues/1006), [#1007](https://github.com/yegor256/cactoos/issues/1007), [#1064](https://github.com/yegor256/cactoos/issues/1064), [#1065](https://github.com/yegor256/cactoos/issues/1065), [#958](https://github.com/yegor256/cactoos/issues/958), [#964](https://github.com/yegor256/cactoos/issues/964), [#980](https://github.com/yegor256/cactoos/issues/980).

**Comment by golszewski86** [comment]
Yeah that was a really brilliant idea @llorllale and @neonailol. Now you can't use Sticky map and list in the same class without using fully qualified name for one of them. And when you see Sticky in the code, you have no idea what it is without looking at import declarations.
Congratulations guys.

**Comment by Vatavuk** [comment]
@golszewski86 You could focus your energy on creating a ticket with detailed explanation on what you consider is wrong and should be changed instead of whining here in the closed PR.

**Comment by llorllale** [comment]
@golszewski86 please open a new ticket and we'll continue there

**Comment by 0pdd** [comment]
@neonailol 3 puzzles [#1003](https://github.com/yegor256/cactoos/issues/1003), [#1025](https://github.com/yegor256/cactoos/issues/1025), [#1046](https://github.com/yegor256/cactoos/issues/1046) are still not solved; solved: [#1004](https://github.com/yegor256/cactoos/issues/1004), [#1005](https://github.com/yegor256/cactoos/issues/1005), [#1006](https://github.com/yegor256/cactoos/issues/1006), [#1007](https://github.com/yegor256/cactoos/issues/1007), [#1064](https://github.com/yegor256/cactoos/issues/1064), [#1065](https://github.com/yegor256/cactoos/issues/1065), [#1074](https://github.com/yegor256/cactoos/issues/1074), [#958](https://github.com/yegor256/cactoos/issues/958), [#964](https://github.com/yegor256/cactoos/issues/964), [#980](https://github.com/yegor256/cactoos/issues/980).

**Comment by neonailol** [comment]
@golszewski86 well, you should read this thread more thoughtfully

**Comment by 0pdd** [comment]
@neonailol the puzzle [#1046](https://github.com/yegor256/cactoos/issues/1046) is still not solved; solved: [#1003](https://github.com/yegor256/cactoos/issues/1003), [#1004](https://github.com/yegor256/cactoos/issues/1004), [#1005](https://github.com/yegor256/cactoos/issues/1005), [#1006](https://github.com/yegor256/cactoos/issues/1006), [#1007](https://github.com/yegor256/cactoos/issues/1007), [#1025](https://github.com/yegor256/cactoos/issues/1025), [#1064](https://github.com/yegor256/cactoos/issues/1064), [#1065](https://github.com/yegor256/cactoos/issues/1065), [#1074](https://github.com/yegor256/cactoos/issues/1074), [#958](https://github.com/yegor256/cactoos/issues/958), [#964](https://github.com/yegor256/cactoos/issues/964), [#980](https://github.com/yegor256/cactoos/issues/980).

**Comment by 0pdd** [comment]
@neonailol all 13 puzzles are solved here: [#1003](https://github.com/yegor256/cactoos/issues/1003), [#1004](https://github.com/yegor256/cactoos/issues/1004), [#1005](https://github.com/yegor256/cactoos/issues/1005), [#1006](https://github.com/yegor256/cactoos/issues/1006), [#1007](https://github.com/yegor256/cactoos/issues/1007), [#1025](https://github.com/yegor256/cactoos/issues/1025), [#1046](https://github.com/yegor256/cactoos/issues/1046), [#1064](https://github.com/yegor256/cactoos/issues/1064), [#1065](https://github.com/yegor256/cactoos/issues/1065), [#1074](https://github.com/yegor256/cactoos/issues/1074), [#958](https://github.com/yegor256/cactoos/issues/958), [#964](https://github.com/yegor256/cactoos/issues/964), [#980](https://github.com/yegor256/cactoos/issues/980).
